VideoGenerator/library/src/main/kotlin/VideoGeneratorImpl.kt

```diff
@@ -56,14 +68,6 @@ class VideoGeneratorImpl(
         recorder.release()
     }
 
-    /**
-     * Finalizes the recording by stopping and releasing the recorder.
-     */
-    protected fun finalize() {
```

Comment:
Any reason you removed this again? Did it not work as expected?

Reply:
I just thought that adding this function would be enough, but the class has to be a closeable, which is some kind of java class. So I don't know how to implement this fully and I removed it again.

Reply:
Yeah, doesnt exactly seem like a destructor now that i look at it. I think asking the user to specifically save the video is fine. I guess one could implement Closable for something like `with (generator = VideoGenerator()) {}`. Reminds me of the python closable